### PR TITLE
Full page alert banner

### LINKED ISSUE #134: Full page alert banner
*Copy-pasting from the related Croydon ticket:*
> When a notable person dies, there is a strict protocol for the council to follow to publicise information about the death.

> A full-page takeover is the recommended response for all users when they visit the site, with a consistent site wide alert present as the user navigates the site.

Pull request: #133

### PR DESCRIPTION
**Pull request by Adnan-cds**
*Full page* bundle for the Alert banner entity.

Contains config files, theme templates, and a related asset library.

This feature is totally dependent on Bootstrap making it unsuitable for general use outside the localgov_theme.  The next step is to replace the Bootstrap specific code with plain Javascript and CSS.

Closes #134 

FAO @andybroomfield @cjstevens78 @tom-steel @willguv

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "homepage": "https://github.com/localgovdrupal/localgov_alert_banner",
     "minimum-stability": "dev",
     "require": {
-        "drupal/field_group": "~3.1"
+        "drupal/field_group": "~3.1",
+        "localgovdrupal/localgov_media": "^1.0@alpha"
```
[the remark this answers is not in the document]
Comment (Adnan-cds) [reply composer.json]: Thanks Andy.

**Comment by Adnan-cds** [comment]
Hi Andy,
A little clarification.  This change request is more of a demo than a merge request.  You may remember I mentioned in Slack that I will pass you the config files we have used for the Full page alert banner.  So here they are.  I don't have any preference for keeping the Full page banner functionality within the localgov_alert_banner module or in one of its submodule or somewhere else.

**Comment by Adnan-cds** [comment]
Hi Andy,
I have turned the Full page alert banner code into a submodule.  Chris is working to replace the Bootstrap-specific parts with plain Javascript and CSS.  Once that's done, we hope to take this change from Draft to Ready-for-review.

@andybroomfield @cjstevens78

**Comment by andybroomfield** [comment]
Thanks @Adnan-cds 
@willguv @finnlewis has this change been approved by product group / working group? Following on from the slack discussions.

**Comment by Adnan-cds** [comment]
Hi Chris,
I have tried to use this on a vanilla LGD site.  It is throwing a Javascript error saying ```favDialog.showModal is not a function```.  I cannot see any code providing ```favDialog```.  My guess is, perhaps a file providing favDialog is not in this repo.  Could you please double check.

@cjstevens78

**Comment by cjstevens78** [comment]
@Adnan-cds ive re-worked this - had a tidy up and fixed the firefox problems.

File: composer.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Adnan-cds) [reply composer.json]: localgovdrupal/localgov_core has now moved into the new ```suggests``` section as per Andy's suggestion.

**Comment by Adnan-cds** [comment]
Ready for review.  Finally :)

To summarize the change, it adds a **submodule** called localgov_alert_banner_full_page that can be enabled or disabled as needed.  When enabled, it offers a bundle (i.e. variant) of the standard Alert banner that can be edited like any other Alert banner.

@andybroomfield @danchamp  @finnlewis  @tom-steel @stephen-cox @willguv

**Review by andybroomfield — COMMENTED** [review]
My glance through at the code says this is fine.
Is there any other council that implements (or is planning on implementing) a full screen alert banner able to test and review this as I'm going to be away.
@danchamp @finnlewis (Lambeth?) or others.

**Comment by finnlewis** [comment]
@Adnan-cds I merged back from the 1.x branch to bring in the latest changes and resolve merge conflict in composer.json.

I am now testing locally and when creating a full page alert, the moderation state form element is displayed on the "Are you sure you want to set live?' page. This seems to prevent me clicking the confirm, so I think the forms are conflicting in some way.

Have you seen this? 

I assume the desired behaviour is to NOT see the moderation state, in line with the default alert banner variant. 

See screenshot below.

image

**Comment by finnlewis** [comment]
@Adnan-cds I fixed that issue with https://github.com/localgovdrupal/localgov_alert_banner/pull/133/commits/941536aef5f12a33c9596a9416a6329c65a36c89

**Comment by finnlewis** [comment]
When logged in, the full page alert is not full page, which is a bit jarring from a UX perspective.

image

I'm guessing with a bit of CSS tinkering we can make it behave the same logged in or logged out.

@Adnan-cds / @cjstevens78  it would be good to get this resolved this sprint, do you have time to look at this today? 

And perhaps @markconroy could you take a quick look at the css / js in this pull request?

**Comment by Adnan-cds** [comment]
> I fixed that issue with 941536a

Thanks Finn :)

> it would be good to get this resolved this sprint, do you have time to look at this today?

I will leave it for Chris.

I like the "Big bad news" image :D

**Comment by finnlewis** [comment]
@cjstevens78 are you able to take a look at this? It would be good to get this merged today or tomorrow if at all possible.... to coincide with the end of sprint. 

If not, if you can point me in the right direction....  :)

**Comment by cjstevens78** [comment]
> @cjstevens78 are you able to take a look at this? It would be good to get this merged today or tomorrow if at all possible.... to coincide with the end of sprint.
> 
> If not, if you can point me in the right direction.... :)

@finnlewis I cant replicate this locally - although thinking about it I wonder if its maybe a good thing for it not to completely hijack the page when logged in because the user in this instance wants to see the site they are interacting with and aren't required to focus their attention on the pop up.

**Comment by cjstevens78** [comment]
@finnlewis we've just had a discussion about this at standup - could we get this merged in as is and look at the logged in overlay as a separate ticket?

**Comment by finnlewis** [comment]
Sure thing, happy to merge as is.

**Review by finnlewis — APPROVED** [review]
Looks good to me!

**Comment by Adnan-cds** [comment]
Thanks Finn.  Merging...

**Pull request merged**